core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/UpxCompressionBuildStep.java

```diff
@@ -34,7 +34,6 @@ public class UpxCompressionBuildStep {
      */
     private static final String PATH = "PATH";
 
-    @BuildStep(onlyIf = NativeBuild.class)
```

Comment:
@dmlloyd I assume this was removed by accident?

Asking because of https://github.com/quarkusio/quarkus/issues/39699

Reply:
If so, #39702 brings it back

Reply:
Yes I believe this must have been an error.